src/tests/test_todo.py

```
@@ -418,18 +523,42 @@ def test_mark_completed_timeout_error(self):
             self.assertIn("Connection timeout", str(context.exception))
-            mock_log_error.assert_called_once()
+            
+            # Verify that error was logged multiple times due to decorators
+            calls = mock_log_error.call_args_list
+            self.assertEqual(len(calls), 4)  # Expect 4 error log calls
+            
+            # Verify the specific error messages
+            self.assertTrue(any(
+                call.args[0] == "Failed to mark task as completed - Error: %s" 
+                and call.args[1] == "Connection timeout"
+                and call.kwargs.get('exc_info') == True
```

Ruff (0.8.2)

Avoid equality comparisons to `True`; use `if call.kwargs.get('exc_info'):` for truth checks

Replace with `call.kwargs.get('exc_info')`

(E712)